src/plugins/unified_search/public/filter_bar/filter_editor/close_confirm_modal.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { memo } from 'react';
+import { EuiConfirmModal } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+interface CloseFilterEditorConfirmModalProps {
+  onCancel: () => void;
+  onConfirm: () => void;
+}
+
+const strings = {
+  getTitle: () =>
+    i18n.translate('unifiedSearch.filter.closeEditorConfirmModal.title', {
+      defaultMessage: 'Unsaved changes',
+    }),
+  getCancelButton: () =>
+    i18n.translate('unifiedSearch.filter.closeEditorConfirmModal.cancelButton', {
+      defaultMessage: 'Cancel',
+    }),
+  getConfirmButton: () =>
+    i18n.translate('unifiedSearch.filter.closeEditorConfirmModal.confirmButton', {
+      defaultMessage: 'Dismiss changes',
+    }),
+  getWarningLabel: () =>
+    i18n.translate('unifiedSearch.filter.closeEditorConfirmModal.warningLabel', {
+      defaultMessage: 'If you leave now, your unsaved work will be lost.',
+    }),
+};
```

Comment:
Can we include more context in this message?
It is too general and can be confused with the other Lens modal about `Unsaved changes`.



<img width="598" alt="Screenshot 2022-12-28 at 10 55 23" src="https://user-images.githubusercontent.com/1421091/209793973-6f8e15b7-cd93-4485-8df8-ab3800ce47e5.png">

>If you leave now,

leave what? is too generic and with a modal that grays out the entire screen is not immediately clear. We were also coming from just a simple popover, not an entire application, so such modal is a bit invasive and can be counterintuitive.
I would rather specify something along those lines: `If you leave now, your unsaved filter changes will be lost.` or something similar, to specify what will be lost.

Also, I've just noticed that we use `Dismiss changes` whereas in Lens modal we use `Discard changes`, probably is good to align on that.

Reply:
cc @andreadelrio, wdyt? 

@markov00 - @andreadelrio is a person, who makes a decision on the text here.

Reply:
cc @KOTungseth for a text check

Reply:
This is a really good call out by @markov00. I think we need to be more specific here.  +1 to `If you leave now, your unsaved filter changes will be lost.` or a modified version of this that @gchaps or @KOTungseth propose.

As an aside, I also noticed that Dashboard has another variation of this `Unsaved changes` popover. I wonder if we should try to align Lens Dashboard with this (i.e. Unified Search Bar).
<img width="441" alt="Unsaved changes" src="https://user-images.githubusercontent.com/4016496/209868513-52f86dd6-0f63-4895-970c-878d24c93f4f.png">

Reply:
@markov00, @andreadelrio, done it. Could you, please, re-review the PR? Thanks.

Reply:
@Kunzetsov I prefer to wait for @gchaps @KOTungseth for the final check.
Still believe that at least we should use the same Lens `Discard changes` text

Reply:
I've also opened an issue on EUI repo about the good practice https://github.com/elastic/eui/issues/6506

Reply:
cc @gchaps @KOTungseth. Could you, please, approve the label? thanks.

Reply:
+1 to being specific about what is being lost.  Also "Discard changes" is the better wording.

Title: Unsaved changes
Description: If you leave now, your unsaved filters will be lost.
Buttons: Cancel | Discard changes

To use elsewhere, replace "filters" with the appropriate noun.